Aquality.Selenium.Core/tests/Aquality.Selenium.Core.Tests/Applications/ApplicationManagerTests.cs

```diff
@@ -13,53 +13,95 @@ public class ApplicationManagerTests
         private const string SpecialSettingsFile = "special";
         private const string SpecialLanguageValue = "special";
         private static readonly TimeSpan SpecialTimeoutValue = TimeSpan.FromDays(1);
+        private const string SpecialLogger = "SpecialLogger";
 
         [Test]
         public void Should_BePossibleTo_RegisterCustomServices()
         {
-            Assert.IsInstanceOf<CustomTimeoutConfiguration>(ApplicationManager.ServiceProvider.GetService<ITimeoutConfiguration>());
+            Assert.IsInstanceOf<TestTimeoutConfiguration>(TestApplicationManager.ServiceProvider.GetService<ITimeoutConfiguration>());
         }
 
         [Test]
         public void Should_BePossibleTo_GetCustomValues()
         {
-            var timeoutConfiguration = ApplicationManager.ServiceProvider.GetService<ITimeoutConfiguration>() as CustomTimeoutConfiguration;
+            var timeoutConfiguration = TestApplicationManager.ServiceProvider.GetService<ITimeoutConfiguration>() as TestTimeoutConfiguration;
             Assert.AreEqual(SpecialTimeoutValue,  timeoutConfiguration.CustomTimeout);
         }
 
+        [Test]
+        public void Should_BePossibleTo_GetCustomLoggerValues()
+        {
+            TestApplicationManager.SetStartup(new CustomStartup());
+            var timeoutConfiguration = TestApplicationManager.ServiceProvider.GetService<ILoggerConfiguration>() as CustomLoggerConfiguration;
+            Assert.AreEqual(SpecialLogger, timeoutConfiguration.CustomLogger);
+        }
+
         [Test]
         public void Should_BePossibleTo_RegisterCustomServices_WithCustomSettingsFile()
         {
-            Assert.AreEqual(SpecialLanguageValue, ApplicationManager.ServiceProvider.GetService<ILoggerConfiguration>().Language);
+            Assert.AreEqual(SpecialLanguageValue, TestApplicationManager.ServiceProvider.GetService<ILoggerConfiguration>().Language);
         }
 
-        private class ApplicationManager : ApplicationManager<IApplication>
+        private class TestApplicationManager : ApplicationManager<IApplication>
         {
-            public static IApplication Application => GetApplication(StartApplicationFunction, () => RegisterServices(services => Application));
+            private static Startup startup = new TestStartup();
```

Comment:
should it be inside the threadlocal container for parallel tests execution?

Reply:
maybe you are right, updated